apps/web-client/src/lib/db/cr-sqlite/__tests__/supplier-order.test.ts

```diff
@@ -170,3 +171,138 @@ describe("Supplier order handlers should", () => {
 		});
 	});
 });
+
+describe("getPlacedSupplierOrderLines", () => {
+	let db: DB;
+	beforeEach(async () => {
+		db = await getRandomDb();
+		// Set up test data
+		await createCustomerOrders(db);
+		// Create two supplier orders using the existing test data
+		await db.exec(`
+			INSERT INTO supplier_order (id, supplier_id, created)
+			VALUES
+			(1, 1, strftime('%s', 'now') * 1000),
+			(2, 2, strftime('%s', 'now') * 1000)
+		`);
+
+		await db.exec(`
+			INSERT INTO supplier_order_line (supplier_order_id, isbn, quantity)
+			VALUES
+			(1, '3', 2),
+			(1, '1', 2),
+			(1, '2', 1),
+			(2, '3', 3)
+		`);
+	});
+
+	it("should retrieve a single supplier order with its line items", async () => {
+		const order = await getPlacedSupplierOrderLines(db, 1);
+
+		expect(order).toEqual([
+			{
+				supplier_id: 1,
+				authors: null,
+				isbn: "1",
+				price: 7,
+				publisher: "MathsAndPhysicsPub",
+				quantity: 2,
+				supplier_order_id: 1,
+				title: "Physics",
+				total_book_number: 5,
+				total_price: 37,
```

Comment:
If this is the "total_book_number" and "total_price" for all lines in this placed supplier order, I'm not sure it makes sense returning it per row in the result?  Similar with "supplier_id", "supplier_name", "supplier_order_id"?

Reply:
I was thinking the less queries the better and the placed order meta data was easy to include with this query